### PR TITLE
Task1

### PR DESCRIPTION
**Pull request by RanaPrathap606**
**Release Notes Description:
Fields:**
**Change Type	Object	Name	Description**
Added	Complaint	Advisory Notice/Recall	Added new Checkbox field
Added	Complaint	Advisory Notice/Recall Number	Added new text field
Added	Complaint	External CAPA Number	Added new text field
Added	Complaint	External ECO/ECR Number	Added new text field
Added	Complaint	External NCMR Number	Added new text field
Added	Complaint	External SCAR Number	Added new text field
Added	Complaint	Investigation Summary	Added new rich text field
Added	Complaint	Legacy Documentation	Added new URL field
Added	Complaint	Legacy Number	Added new text field
Added	Complaint	NCMR	Added new checkbox field
Added	Complaint	NCMR Number	Added new field with lookup to Nonconformance
Added	Complaint	Results of Investigation	Added new picklist field with below values
Conforms
Nonconformity
Cannot Determine
Added	Complaint	SCAR	Added new text field
Added	Complaint	Supplier Escalation	Added new field with lookup to Supplier Escalation
Modified	Complaint	ECO/ECR	Updated Field Label from Change Control to ECO/ECR using Overrides in Translation
Modified	Complaint	ECO/ECR Number	Updated Field Label from Change Control Number to ECO/ECR Number using Overrides in Translation

**Custom Object Translations:**
Change Type	Object	Name	Description
Modified	Complaint	ECO/ECR	Updated Field Label from Change Control to ECO/ECR using Overrides in Translation
Modified	Complaint	ECO/ECR Number	Updated Field Label from Change Control Number to ECO/ECR Number using Overrides in Translation

**FLOWS:**
Change Type	Name	Description
Created	CQ_TASK_Investigation_After_Update_RT_Flow	Added new After Update Record Trigger Flow for Investigation.
Created	CQ_TASK_Complaint_OnInvestigationComplete_SubFlow	Added new Sub Flow for update of Complaint when Investigation is completed.

**LWC:**
Change Type	Name	Description
Created	cquiTaskComplaintEdit	Created the LWC Layout to Add the fields mentioned in Closure Section.
Page Layout:
Change Type	Object	Name	Description
Created	Complaint	CQ TASK Complaint Layout	Updated the Page Layout to add the fields mentioned.

**Permission Set:**
Change Type	Name	Description
Created	CQ TASK Complaint Permission Set	Created new Permission Set related to Complaint Object and added permission to created fields.

### CODE REVIEW

**Comment by RanaPrathap606** [comment]
Rabin, please review the Pull Request.

File: Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/**
+ * **************Revision History*******************
+ * 1. [Task 1]: Given Investigation Record Status is updated to Complete then
+ *              Complaint Investigation Summary and Result of Investigation are updated.
+ * Flow : CQ_TASK_Investigation_After_Update_RT_Flow
+ *        CQ_Complaint_OnInvestigationComplete_SubFlow
```
Comment (rabinshrestha130) [inline Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls]: Update the flow name

File: Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+ * --------------------------------------------------
+ * 
+**/
+@isTest public with sharing class CQ_TASK_Test_Complaint_UpdateFlowTest {
```
Comment (rabinshrestha130) [inline Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls]: after @isTest leave the line

**Review by rabinshrestha130 — CHANGES_REQUESTED** [review]

File: Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls [hunk 3]
```diff
@@ -0,0 +1,121 @@
+/**
```
Comment (rabinshrestha130) [inline Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls:1]: where is the meta file for new added class?

File: Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    /**
+     * GIVEN: A Investigation Record
+     * WHEN : Status = Completed
+     * THEN : Update Investigation Summary and Result of Investigtion fields on Complaint 
+     *          from Investigation Summary and Conclusion fields.
+     * 
+     * Flow : CQ_TASK_Investigation_After_Update_RT_Flow
+     *        CQ_Complaint_OnInvestigationComplete_SubFlow
```
Comment (rabinshrestha130) [inline Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls]: update flow name

File: Main/Src/objects/compliancequest__SQX_Complaint__c/fields/CQ_TASK_Investigation_Summary__c.field-meta.xml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>CQ_TASK_Investigation_Summary__c</fullName>
+    <description>This is a read only text field that will get auto-populated with information entered in Investigation Summary of the linked Investigation record.</description>
+    <externalId>false</externalId>
+    <label>Investigation Summary</label>
+    <length>32768</length>
+    <trackFeedHistory>false</trackFeedHistory>
+    <trackHistory>false</trackHistory>
+    <trackTrending>false</trackTrending>
+    <type>Html</type>
```
Comment (rabinshrestha130) [inline Main/Src/objects/compliancequest__SQX_Complaint__c/fields/CQ_TASK_Investigation_Summary__c.field-meta.xml:11]: why is this type html?

File: Main/Src/objects/compliancequest__SQX_Complaint__c/fields/CQ_TASK_Legacy_Documentation__c.field-meta.xml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>CQ_TASK_Legacy_Documentation__c</fullName>
+    <description>Custom Checkbox field for Legacy Documentation.</description>
+    <externalId>false</externalId>
+    <label>Legacy Documentation</label>
+    <required>false</required>
+    <trackFeedHistory>false</trackFeedHistory>
+    <trackHistory>false</trackHistory>
+    <trackTrending>false</trackTrending>
+    <type>Url</type>
```
Comment (rabinshrestha130) [inline Main/Src/objects/compliancequest__SQX_Complaint__c/fields/CQ_TASK_Legacy_Documentation__c.field-meta.xml:11]: why is type url?

File: Main/Src/objects/compliancequest__SQX_Complaint__c/fields/CQ_TASK_SQX_Supplier_Escalation__c.field-meta.xml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>CQ_TASK_SQX_Supplier_Escalation__c</fullName>
+    <deleteConstraint>SetNull</deleteConstraint>
+    <description>This field functions the link/lookup a Supplier Escalation record.</description>
+    <externalId>false</externalId>
+    <label>Supplier Escalation</label>
+    <referenceTo>compliancequest__SQX_Supplier_Escalation__c</referenceTo>
+    <relationshipLabel>Complaints</relationshipLabel>
+    <relationshipName>CQ_CQ_SQX_Complaints</relationshipName>
```
Comment (rabinshrestha130) [inline Main/Src/objects/compliancequest__SQX_Complaint__c/fields/CQ_TASK_SQX_Supplier_Escalation__c.field-meta.xml:10]: follow naming convention

**Review by rabinshrestha130 — CHANGES_REQUESTED** [review]

File: Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls (hunk 1 above)
Comment (RanaPrathap606) [reply Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls]: Updated Flow Name to  CQ_TASK_Complaint_OnInvestigationComplete_SubFlow.

File: Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls (hunk 2 above)
Comment (RanaPrathap606) [reply Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls]: Left line after @isTest.

File: Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls (hunk 4 above)
Comment (RanaPrathap606) [reply Main/Src/classes/CQ_TASK_Test_Complaint_UpdateFlowTest.cls]: Updated floe name to CQ_TASK_Complaint_OnInvestigationComplete_SubFlow

**Pull request merged**